addOns/client/src/main/java/org/zaproxy/addon/client/spider/ClientSpider.java

```diff
@@ -312,22 +330,45 @@ public void eventReceived(Event event) {
             if (ClientMap.MAP_COMPONENT_ADDED_EVENT.equals(event.getEventType())) {
                 if (ClickElement.isSupported(href -> href.startsWith(targetUrl), parameters)) {
                     addTask(
+                            url,
                             openAction(
                                     url,
                                     new ClickElement(valueProvider, createURI(url), parameters)),
-                            options.getPageLoadTimeInSecs());
+                            options.getPageLoadTimeInSecs(),
+                            Constant.messages.getString("client.spider.panel.table.action.click"),
+                            paramsToString(parameters));
                 } else if (SubmitForm.isSupported(parameters)) {
                     addTask(
+                            url,
                             openAction(
                                     url, new SubmitForm(valueProvider, createURI(url), parameters)),
-                            options.getPageLoadTimeInSecs());
+                            options.getPageLoadTimeInSecs(),
+                            Constant.messages.getString("client.spider.panel.table.action.submit"),
+                            paramsToString(parameters));
                 }
             } else {
                 addOpenUrlTask(url, options.getPageLoadTimeInSecs());
             }
         }
     }
 
+    private static String paramsToString(Map<String, String> parameters) {
+        String tag = parameters.get("tagName");
+        if (tag != null) {
+            switch (tag) {
+                case "A":
+                    return Constant.messages.getString(
+                            "client.spider.panel.table.details.link",
+                            parameters.get("href"),
+                            parameters.get("text"));
+                case "BUTTON":
+                    return Constant.messages.getString(
+                            "client.spider.panel.table.details.button", parameters.get("text"));
```

Comment:
No need for it to block this, but it seems we should have constants for a bunch of field/item names.